### ISSUE TITLE
How are we doing so far?

### ISSUE THREAD

**Issue by docwilmot** [backdrop-ops/contrib#34]
Drupal 7.x modules list by most installed (page 1):
(thought it would be nice to see what's missing - we're not doing too bad!)

- [ ] Views
- [ ] Chaos tool suite (ctools)
- [ ] Token
- [ ] Pathauto
- [ ] Libraries API
- [ ] Administration menu
- [ ] Entity API
- [ ] Webform
- [ ] IMCE
- [ ] Date
- [ ] Wysiwyg
- [ ] Google Analytics
- [ ] CKEditor - WYSIWYG HTML editor
- [ ] Link
- [ ] jQuery Update
- [ ] Backup and Migrate
- [ ] Views Slideshow
- [ ] Rules
- [ ] XML sitemap
- [ ] CAPTCHA
- [ ] Module Filter
- [ ] Panels
- [ ] Colorbox
- [ ] Features
- [ ] Views Bulk Operations (VBO)

**Comment by oadaeh** [comment]
Webform? Nobody users Webform.

**Comment by quicksketch** [comment]
@oadaeh :wink:

I'll work on porting it after we get some of these critical infrastructure modules fully converted. I'd like to get our main site off of Drupal 7 asap, so I'll probably prioritize that first.

CTools I think we could mark off. We have the following pieces of CTools in core:

- Dropbuttons
- Evalmath
- Dialogs
- Exportability (via CMI)
- Tempstore

The primary piece of CTools we do not have is the concept of a plugin, which means combining the "discovery" mechanism with the code implementation. Instead we've standardized on using _info() hooks for discovery. e.g. `hook_blocks_info()`, `hook_views_data()`, etc. Using plugins would essentially mean we'd rewrite half of core to eliminate info hooks, which could send us down the same rewriting rabbit hole Drupal 8 went down. Some historical discussion at https://github.com/backdrop/backdrop-issues/issues/41

So... crossed off CTools. :smile:

**Comment by quicksketch** [comment]
Entity API is also an interesting one, because we've converted Entities into "real" objects. We integrated about as much of it as we'd want in core. I'm not sure a contrib version of that module should even exist for Backdrop in the first place. See the meta at https://github.com/backdrop/backdrop-issues/issues/49.

Crossed off Entity API.

**Comment by Graham-72** [comment]
Is there a list of the current status of Backdrop versions? Are they all 'under development' at the moment?

**Comment by quicksketch** [comment]
I think most all of them are "under development". I tagged a few stable releases for External Links and Devel module, but I don't think many projects have releases yet. I don't think Github makes it easy to get a list of official releases within an entire group. Yet another reason why we need to get backdropcms.org up and running with a contrib listing.

**Comment by oadaeh** [comment]
@quicksketch, you definately are going to need some help with this.  Between your day job, keeping up with Backdrop core and the contrib requests that seem to be coming in fairly quickly, you are going to be stretched thin.  I suspect your wife might want some of your attention, too.

**Comment by quicksketch** [comment]
Hi @oadaeh! Ha, I completely agree. No one could manage a Drupal-sized project alone. We added an additional contrib reviewer (klonos) just yesterday: https://github.com/backdrop-ops/contrib/issues/45

The contrib group itself is a great way to spread out responsibility. Already with all these new developers coming in, it'll reduce the effort Jen and I have to spend on individual modules. I'm not sure what we'll do about Webform in the short term. I'd love if another developer were to make the initial port but add me as a maintainer (as happened with Insert in https://github.com/backdrop-ops/contrib/issues/37).

@oadaeh if you're interested in helping review applications as well, I'd be more than happy to add you to the contrib management group as well. :smile:

**Comment by klonos** [comment]
> ...you are going to be stretched thin.

I propose changing your nickname to "quickstretch" then :p

Naughty me! Don't have enough time in my day and still find time to post silly things. Sorry, couldn't help it.

**Comment by oadaeh** [comment]
@klonos Ha!
@quicksketch nice try, but I guess I asked for it by opening my mouth. I'm going to be challenged enough with scraping out the time to port the list I have on my plate.

**Comment by quicksketch** [comment]
:wink: Well ports are still moving on.

The folks at CivicActions apparently have taken a stab at porting CKEditor:
http://www.civicactions.com/blog/2015/feb/02/first-look-backdrop-cms, we might be able to get them to move that into contrib.

@docwilmot also is 90% through porting Link module: https://github.com/docwilmot/link. There's just a caveat that it's not actually saving data yet. :wink:

No one is working on Date yet that I know of.

Libraries API we should work on integrating in core (or parts of it), see https://github.com/backdrop/backdrop-issues/issues/159. Also looks like @oadaeh has filed an issue at https://www.drupal.org/node/2417985. For the time being, it looks like several contrib projects have just started bundling the libraries with their modules, since we're not bound by Drupal.org's licensing restrictions. I think that's fine for most projects honestly. Bundled libraries are *so much easier* for end-users, though it requires more effort for contrib modules to be updating the bundled libraries all the time.

**Comment by docwilmot** [comment]
See https://github.com/backdrop-ops/contrib/issues/50. Link seems to be working there.

**Comment by klonos** [comment]
Hey, I just wanted to update the issue summary with links to respective projects in backdrop contrib etc., but it seems I don't have the permission to do so. Hate to be busting people's b@@ls, but this was so much easier on d.org.

Do you think it would be a good idea to have this summary some where in bd.org? I like the title: "Porting the most popular Drupal modules to Backdrop - How are we doing so far?" I think it would be good for publicity + it would get people to try them out and report any issues.

**Comment by sirkitree** [comment]
Helping to publicize this one on the forum: http://www.reddit.com/r/backdrop/comments/2xx6ix/porting_the_most_popular_drupal_modules_to/

**Comment by biolithic** [comment]
Hey, don't forget about https://github.com/biolithic/backdrop_colorbox :)

**Comment by biolithic** [comment]
I would like to tackle Views Slideshow and XML Sitemap this Sunday (if no one else), leaving Rules and Date as the only real modules left on this list.  Where's the second page list?
Depending on time, could probably start porting Webform if need be, then leave big decisions to the BDFL for it

**Comment by Graham-72** [comment]
I am currently working on a port of Rules. I have done a port of CKEditor for use until it is in core, so perhaps the list could be updated?

**Comment by quicksketch** [comment]
Sure, this list is pretty informal anyway. Updated. :)

**Comment by quicksketch** [comment]
I added the second page, now this is a list of top 50 modules. Still looking pretty good. :smile:

**Comment by IbnSaeed** [comment]
Can Drupal Commerce be ported ?

**Comment by Graham-72** [comment]
I am proposing to start porting Date in the next day or so.

**Comment by biolithic** [comment]
I am starting to work on porting Webform today now that I'm waiting on
something

**Comment by biolithic** [comment]
Gee, that was sure an easy port of Webform - nothing broke after making two webforms with random options.  I know I have to do lots of testing of the options but I wonder if there is a connection between the owner of the module and Backdrop.

**Comment by pablo-romero** [comment]
What about [Metatag](https://www.drupal.org/project/metatag) module?
I'm surprised it's not in the list, since it's one I use most of the times in D7.

**Comment by docwilmot** [comment]
@pablo-romero this was probably missed in the cutting and pasting. I've added it.

**Comment by klonos** [comment]
> ...I wonder if there is a connection between the owner of the module and Backdrop.

:wink:

@quicksketch Hey Nate, still don't have enough time in my life, but I can still contribute small things when I find the energy. Can you look into allowing me to edit issues in this repo so I can edit issue summaries? Thanx.

**Comment by quicksketch** [comment]
> Can you look into allowing me to edit issues in this repo so I can edit issue summaries? Thanx.

Sure, I resent a group invitation.

**Comment by klonos** [comment]
Thanx, Just joined. Going straight to bed though. Will contribute once I make some time. :sleeping:

**Comment by klonos** [comment]
Just updated the issue summary with links to the roadmap page and what was done in 1.2.0. Perhaps there's a few issues here and there filed about getting some things in core, but don't have more time now.

**Comment by danielrose28** [comment]
I've been posting in gitter, but thought I'd copy my post in here as well:

I’m going to try to port the [IMCE Mkdir](https://www.drupal.org/project/imce_mkdir) module. I also created an [issue](https://www.drupal.org/node/2690795) in the project's issue queue for it. Hopefully that looks ok. I think this would be a good one to get my feet wet so to speak.

**Comment by MrHaroldA** [comment]
I'm trying to pitch Backdrop for a new project here, which will need: Domains, Feeds, Views Data Export, Field Groups, and probably a lot more ;)

I've already started on a D7 Domains-like module, based on (fieldable) entities. Feeds should be ported already, but Views Data Export and Field Groups might take up too much time ... and word on porting those?

**Comment by biolithic** [comment]
Hi MrHaroldA,
I've been a little busy with work and other BD things (layouts, responsive
images, trying to write a blog post(s)) that I have kinda slacked off on
the module porting.  Field Group will be easier than Field Collection, but
I wonder if it should be broken into sub-modules just to make progress on
it if its needed soon.

Otherwise, interested in the pitch and the community response!

**Comment by richie-harris-smc** [comment]
I've ported the PHP filter module that was removed from core. (was really simple) Don't see a reason it can't be a contrib mod. I understand the argument not to have it in core, but as contrib seems fine.

I've also started porting Commerce. Have Commerce, Commerce Product, and Commerce Price done. Now going through the rest of the sub mods, then will look to tackle Commerce Shipping. Commerce Shipping seems like it might be more work than Commerce itself...as I believe it depends on CTools plugins.

**Comment by quicksketch** [comment]
Hi @richie-harris-smc! Welcome to the Backdrop community! Once you're ready, we'd love to have you share your modules in the official [Backdrop Contrib group](https://github.com/backdrop-contrib) (which also syncs with BackdropCMS.org). When you're ready, you can file an issue to have the module reviewed at https://github.com/backdrop-ops/contrib

**Comment by thamas** [comment]
@richie-harris-smc have you seen https://github.com/codewombat/backshop? (It was mentioned as an answer to my question: https://github.com/backdrop/backdrop-issues/issues/1831#issuecomment-216542259) Maybe it would be better to join forces than to work on two similar solutions.

**Comment by richie-harris-smc** [comment]
@thamas I hadn't seen that, but I don't know that that's a solution for our clients. We have a number of clients with highly customized shopping carts that will want an upgrade path from D7 once it'd EOL'd. They'll want something that doesn't involve starting over. The work I'm doing to port Commerce is to the purpose of providing a direct upgrade path for existing D7 Commerce users. I'm not far enough along to contribute it as a project, but expect to be in the next few weeks. If Backshop has some traction by then, I would understand if the community decided they weren't interested in my solution. But it's something I'm going to end up doing anyway.

**Comment by klonos** [comment]
@richie-harris-smc ++ on this matter. There is *a lot* of documentation out there for Commerce and also a big audience of people that would possibly need a Backdrop version (all the current D7 Commerce users actually). These people have invested a lot of time, energy and resources and I am sure that they would like to have some continuation. It is within the spirit of the [Backdrop philosophy](https://backdropcms.org/philosophy) to provide a solution that would minimize cost for them. One of our long-used motos was "[There is a cost that comes with change](http://www.jenlampton.com/presentations/backdrop-cms-theres-cost-comes-change-2)..."

**Comment by thamas** [comment]
@richie-harris-smc @klonos Ok I understand your motivations. I do not know more about Backshop just what is published on its project page. I also understand that while you are working actively Backshop development stagnates.
 
I just wanted you to know about it because I could imagine that there would be a simple upgrade path from D7 Commerce to Backshop too. Also I thought it is possible that there are work done in both projects which would be useful for each other. (These are all supposition I did not check the code.)

**Comment by rszrama** [comment]
Thanks for the ping. I wouldn't currently support a straight port of Commerce 1.x, but obviously don't mind anyone doing it for their own sake. My goal w/ Backshop is to combine the best of Commerce 1.x and 2.x with a philosophy that fits Backdrop's target audience. We can always advise on a Commerce port, though.

**Comment by thamas** [comment]
Thanks for the answer and for the info @rszrama!

**Comment by vonnnew** [comment]
Hi all,  I'm new to Backdrop and I'm working on a project that requires Field Collection.  If no one else is working on it, I will attempt a port.  It's my first.

**Comment by vonnnew** [comment]
Can someone point me to a forum or channel where I might be able to get some mentoring on porting D7 modules?  I'm trying to port Field Collections and I have a couple of others that I will work on if I get that one, but I'm hitting a wall.  All the documentation I've been able to find so far has sort of a 'now make it work' step in the middle.  I don't want to clutter this thread with my newby questions.

**Comment by MrHaroldA** [comment]
You may use the Gitter chat for newby question: https://gitter.im/backdrop

**Comment by biolithic** [comment]
https://gitter.im/backdrop/backdrop-issues ?

**Comment by jackbravo** [comment]
@richie-harris-smc how are you doing with the commerce backport to backdrop (back back back :-p). We are considering using backdrop for a proyect, but we need commerce and we could help with the porting.

**Comment by oadaeh** [comment]
Joaquin, did you check out Ryan's work at https://github.com/codewombat/backshop ?

**Comment by rszrama** [comment]
fwiw, I haven't touched that in a long while - taking over Commerce Guys became my nights / weekends hobby project. I haven't lost hope, but it's back burner atm and was dependent on ongoing contribution to the core Entity API. (I also didn't really want to port Entity Reference myself, but that's what it needs. ; )

**Comment by Al-Rozhkov** [comment]
Looks like this list needs to be updated. I saw releases for Profile and Better exposed filters in backdrop-contrib. Probably there are ports of other modules from this list too.

**Comment by docwilmot** [comment]
Pinging @richie-harris-smc

**Comment by JAyenGreen** [comment]
Has anyone thought of tackling (or begun tackling) Apache Solr and Facet API modules?

**Comment by jlfranklin** [comment]
Views Data Export can be checked off this list.  I think there are several others listed here that have been ported.

**Comment by MrHaroldA** [comment]
>  I think there are several others listed here that have been ported.
I also think some of them should be put back on the list, like Webform. Porting is only the start, maintaining is mandatory.

For example, I've built a website this weekend which evolves around 2 forms, so I installed Backdrop + Webform. I built one form which has around 20 fields and fieldsets, and started working on the conditionals. I soon had Javascript errors when conditionally showing a fieldset. After some searching, this is fixed in Drupal's Webform, as well as many other fixes, around 4.5/4.6.

So I deleted Backdrop and installed Drupal 7. Which feels like a big step back in time, but Webform works perfectly. Backdrop's Webform is a fork of 4.4, according to it's readme. I installed 4.14 on Drupal, which is also a half year old.

This even made me rethink using Backdrop.

**Comment by klonos** [comment]
@MrHaroldA I understand that when things need to get done they need to get done, so building something with D7 is unavoidable, but I have to also point that maintaining is as mandatory as reporting issues and helping getting them fixed.

**Comment by MrHaroldA** [comment]
> maintaining is as mandatory as reporting issues and helping getting them fixed.

This is the backlog for Webform: https://github.com/backdrop-contrib/webform/issues/39

That's not something I can do with my very limited spare time. In fact, it might be quicker to port D7-4.14 instead of merging all patches separately.

But this is only Webform. I believe that all projects merged into Backdrop are in great shape, but I doubt that the ported contribs are.

**Comment by TonyTheFerg** [comment]
> @thamas I hadn't seen that, but I don't know that that's a solution for our clients. We have a number of clients with highly customized shopping carts that will want an upgrade path from D7 once it'd EOL'd. They'll want something that doesn't involve starting over. The work I'm doing to port Commerce is to the purpose of providing a direct upgrade path for existing D7 Commerce users. I'm not far enough along to contribute it as a project, but expect to be in the next few weeks. If Backshop has some traction by then, I would understand if the community decided they weren't interested in my solution. But it's something I'm going to end up doing anyway.

@richie-harris-smc have you made any more progress on a 1:1 port of commerce 1?

**Comment by docwilmot** [comment]
@TonyTheFerg I passed Commerce through Coder Upgrade and fixed a few basic things just to get it to install without error. Its here: https://github.com/backdrop-contrib/commerce

I had reached out to @rszrama here https://github.com/codewombat/backshop/issues to let him know. 

Thats about as much as I intended to do there. I dont feel confident I know enough about something as touchy as ecommerce to finish this., but very happy to help anyone who wants to take it up again and get the bugs squashed.

**Comment by cptX** [comment]
What about porting the **advanced forum** module? I think 3 people already requested this almost a year already. Tried to use forum-ng but is not in production state. Other person tried to build a forum with views. Personally I'm still in development. This lack of advanced forum module got me several delays in my dev. Currently I'm using the existing forum module but I think is not sufficient for a respectful forum.

**Comment by richardlampitt** [comment]
> What about porting the **advanced forum** module? I think 3 people already requested this almost a year already. Tried to use forum-ng but is not in production state. Other person tried to build a forum with views. Personally I'm still in development. This lack of advanced forum module got me several delays in my dev. Currently I'm using the existing forum module but I think is not sufficient for a respectful forum.

I've replied in issue [issue #475](https://github.com/backdrop-ops/contrib/issues/475), but I'll do so here for the record as well - I might be porting this module at some point in the near future as part of work for a client.

**Comment by richardlampitt** [comment]
Our client has given us the go-ahead for this project, so I'm now working on porting the `advanced_forum` module at this very moment.

**Comment by argiepiano** [comment]
While the checkboxes in the OP list Views Bulk Operations being in core, this is not quite accurate. The VBO version  in core has different code and fewer features than the D7 contrib module. Also, none of the VBO views fields from D7 will transfer to Backdrop when upgrading. 

There IS a stable [contrib version](https://github.com/backdrop-contrib/views_bulk_operations) of D7's VBO that should be listed above. Unfortunately, for some reason, I've lost the ability to edit that.

**Comment by olafgrabienski** [comment]
@argiepiano I've added the link to the Backdrop contrib VBO module in the list.
